examples/qualcomm/scripts/utils.py

```diff
@@ -212,25 +228,39 @@ def build_executorch_binary(
         skip_node_id_set,
         skip_node_op_set,
     )
-    edge_prog.exported_program = to_backend(edge_prog.exported_program, qnn_partitioner)
-    edge_prog.exported_program.graph_module.graph.print_tabular()
-    exec_prog = edge_prog.to_executorch(
-        config=ExecutorchBackendConfig(
-            extract_constant_segment=False,
-            # For shared buffer, user must pass the memory address
-            # which is allocated by RPC memory to executor runner.
-            # Therefore, won't want to pre-allocate
-            # by memory manager in runtime.
-            memory_planning_pass=MemoryPlanningPass(
-                memory_planning_algo="greedy",
-                alloc_graph_input=not shared_buffer,
-                alloc_graph_output=not shared_buffer,
-            ),
-            extract_delegate_segments=True,
-        )
+
+    executorch_config = ExecutorchBackendConfig(
+        extract_constant_segment=False,
+        # For shared buffer, user must pass the memory address
+        # which is allocated by RPC memory to executor runner.
+        # Therefore, won't want to pre-allocate
+        # by memory manager in runtime.
```

Comment:
This seems like an optmization opportunity and we can add them later? For the alpha release, maybe let's get a functional version, and improve it step by step. What do you think?

Reply:
The shared buffer has already been done by https://github.com/pytorch/executorch/pull/2531. Shared buffer mechanism is enabled for the rest of examples but not llama now (we'll have it in the near future, need changes in `qnn_llama_runner`).
For now, we still need this refactored code for other examples being functional as usual.